commit 31d8a958610592bce683366c0808fb610f67a366
Author: Guido Draheim <guidod@gmx.de>
Date:   Sun Apr 19 18:25:16 2020 +0200

    copy directory instead of unpacking manpages.tar to install-man #93

diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index 623da24..b679f7b 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -174,33 +174,28 @@ add_custom_target(install-site
    DEPENDS site.html zziplib.html zzipmmapped.html zzipfseeko.html
    VERBATIM)
 add_custom_target(install-htmpages
-   COMMAND ${BASH} -c "ls -l ${CMAKE_CURRENT_BINARY_DIR}/*pages.tar"
    COMMAND ${BASH} -c "mkdir -vp $DESTDIR${docdir}  >&2 || ls -ld $DESTDIR${docdir} >&2"
-   COMMAND ${BASH} -c "cd $DESTDIR${docdir} && tar x -f ${outdir}/htmpages.tar"
+   COMMAND ${BASH} -c "cp -r ${outdir}/html $DESTDIR${docdir}/"
    COMMAND ${BASH} -c "cd $DESTDIR${docdir} && mv html man"
    DEPENDS htmpages.tar
    VERBATIM)
 add_custom_target(install-manpages
-   COMMAND ${BASH} -c "ls -l ${CMAKE_CURRENT_BINARY_DIR}/*pages.tar"
    COMMAND ${BASH} -c "mkdir -vp $DESTDIR${mandir}  >&2 || ls -ld $DESTDIR${mandir} >&2"
-   COMMAND ${BASH} -c "cd $DESTDIR${mandir} && tar xf ${outdir}/manpages.tar"
+   COMMAND ${BASH} -c "cp -r ${outdir}/man3 $DESTDIR${mandir}/"
    DEPENDS manpages.tar
    VERBATIM)
-
+if(ZZIP_HTMPAGES)
+   install(CODE "message(STATUS \"Installing: html pages to ${docdir}/man\")")
+   install(DIRECTORY html/ DESTINATION ${docdir}/man MESSAGE_NEVER)
+endif()
+if(ZZIP_MANPAGES)
+   install(CODE "message(STATUS \"Installing: man3 pages to ${mandir}/man3\")")
+   install(DIRECTORY man3/ DESTINATION ${mandir}/man3 MESSAGE_NEVER)
+endif()
 add_custom_target(install-docs DEPENDS install-manpages install-htmpages install-site)
 add_custom_target(install-mans DEPENDS install-manpages)
-
-if(ZZIP_MANPAGES)
-add_custom_target(install-man ALL DEPENDS install-manpages)
-else()
 add_custom_target(install-man DEPENDS install-manpages)
-endif()
-
-if(ZZIP_HTMPAGES)
-add_custom_target(install-htm ALL DEPENDS install-htmpages)
-else()
 add_custom_target(install-htm DEPENDS install-htmpages)
-endif()
 
 # messages #######################################################
 
